extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -102,7 +119,7 @@ public void initialize() {
         }
 
         DtlsConnectorConfig.Builder builder = new DtlsConnectorConfig.Builder(new InetSocketAddress(0));
-        builder.setPskStore(new StaticPskStore("", configuration.code.getBytes()));
+        builder.setPskStore(new StaticPskStore(configuration.username, configuration.userkey.getBytes()));
```

Comment:
How do we notice if there is any authentication failure with these values? The user might have resetted the gateway or the gateway might have lost the user credentials through some other way. If they are not valid, we must set the Thing status to OFFLINE CONFIGURATION_ERROR.

Reply:
From my experience with this protocol - there is NO way to identify authentication failure. The only place where the errors are handled is the ```CoapHandler.onError``` but the error can be any: not authenticated, no connection etc., and the only possible handling is to update the status to ```ThingStatusDetail.COMMUNICATION_ERROR``` which is actually done